### PR TITLE
Extract replicaset member creation logic

### PR DESCRIPTION
**Pull request by mateusduboli**
This is related to #165, it's a simple refactor to ease understanding of the code to be changed in the other issue.

My argument is that the replicaset member configuration should be constant for every run, with the same set of attributes.

This will make it easier to make sense of the replicaset code

### CODE REVIEW

**Comment by mateusduboli** [comment]
@josephholsten Could you take a look on this?

**Comment by mateusduboli** [comment]
Sorry! Forgot to push some modifications.

**Comment by mateusduboli** [comment]
Done!

**Comment by mateusduboli** [comment]
For what I'm investigating this errors and incompatibility issues caused by the latest `chefdk` version release.

In my setup both of this commands worked without any issues.
`$ chef exec rake`
`$ kitchen test all`

The issue, and that's can either be faced as a CI issue as an environment one, is that the version that `brew` provides is fixed on the latest stable of chefdk `1.5.x` and the CI uses the latest development version of it `2.0.x`.

I don't know if I should update, since this is probably the most used version in the wild, or the CI should test for this release as well.

File: libraries/mongodb.rb [hunk 1]
```diff
@@ -45,31 +74,9 @@ def self.configure_replicaset(node, name, members)
...
-                 end
-      rs_options[host]['priority'] = priority unless priority == 1
-      tags = members[n]['mongodb']['replica_tags'].to_hash
-      rs_options[host]['tags'] = tags unless tags.empty?
-      votes = members[n]['mongodb']['replica_votes']
-      rs_options[host]['votes'] = votes unless votes == 1
-      rs_members << { '_id' => n, 'host' => host }.merge(rs_options[host])
+
+    rs_members = members.each_with_index.map do |member, n|
+      create_replicaset_member(member).merge('_id' => n)
```
Comment (josephholsten) [inline libraries/mongodb.rb]: so clean!

File: libraries/mongodb.rb [hunk 2]
```diff
@@ -80,17 +87,11 @@ def self.configure_replicaset(node, name, members)
...
-      port = members[n]['mongodb']['config']['port']
-      rs_member_ips << { '_id' => n, 'host' => "#{members[n]['ipaddress']}:#{port}" }
-    end
-
     admin = connection['admin']
     cmd = BSON::OrderedHash.new
     cmd['replSetInitiate'] = {
       '_id' => name,
-      'members' => node['mongodb']['use_ip_address'] ? rs_member_ips : rs_members,
+      'members' => rs_members,
```
Comment (josephholsten) [inline libraries/mongodb.rb]: yay for killing the IP address approach

Comment (mateusduboli) [inline libraries/mongodb.rb]: I'm actually not killing it :P
I moved this part of the code to the [extracted function](https://github.com/mateusduboli/mongodb/blob/f40f24c888bc84fbd455e0e91291821119e22b05/libraries/mongodb.rb#L29), I want to do a little more but would rather split in different PRs.

I also used the same variable to set ports, since on the old code there were 2 variables for it.

File: libraries/mongodb.rb [hunk 3]
```diff
@@ -22,6 +22,35 @@
 require 'json'
 
 class Chef::ResourceDefinitionList::MongoDB
+  def self.create_replicaset_member(node)
```
Comment (mateusduboli) [inline libraries/mongodb.rb]: Only the `mongodb` parts or also the `ohai` variables like `ipaddress`?

Comment (josephholsten) [inline libraries/mongodb.rb]: All of them, because at some point we'd like to be able to support non-chef service discovery. I don't think we need to put a lot of effort into defining the field meanings, just specifying which ones are required, which are optional, and what the defaults are.

File: libraries/mongodb.rb (hunk 2 above)
Comment (josephholsten) [inline libraries/mongodb.rb]: ah, shame on me for skimming to quickly!

and I was thinking of #165, which actually is terrible.

But yay for consistency and handling related things in the same place!

**Comment by damacus** [comment]
@shortdudey123 I think this one is ready to merge now. 

Figured I'd ping you considering you've done a lot of work on this recently 👍

File: libraries/mongodb.rb (hunk 3 above)
Comment (mateusduboli) [inline libraries/mongodb.rb]: Which documentation style should I use?
I was trying to rdoc but the docs are really lackluster, if you could point me out to a good source that would be of great help :)

File: libraries/mongodb.rb (hunk 2 above)
Comment (mateusduboli) [inline libraries/mongodb.rb]: I'll close #165 then, but I still need that functionality since I uploaded the cookbook with that patch in order to my database to work

**Comment by mateusduboli** [comment]
@shortdudey123 should we merge this, after the aprovais?

**Comment by mateusduboli** [comment]
One other suggestion, since I don't have the habili

**Comment by mateusduboli** [comment]
Sorry, didn't meant to close the PR
